components/eventing-controller/controllers/subscription/nats/integration_test.go

```diff
@@ -0,0 +1,837 @@
+				}
+			},
+			expectAfter: expect{
+				k8sSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.HaveCondition(reconcilertesting.DefaultCondition("")),
+					reconcilertesting.HaveSubsConfiguration(
+						&eventingv1alpha1.SubscriptionConfig{
+							MaxInFlightMessages: 101,
+						},
+					),
+				},
+				natsSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.BeNotNil(),
+					reconcilertesting.HaveSubject(cleanEventType("")),
+					reconcilertesting.BeValid(),
+				},
+			},
+			shouldTestDeletion: true,
+		},
+
+		{
+			name: "resolve multiple conditions",
+			subscriptionOpts: []reconcilertesting.SubscriptionOpt{
+				reconcilertesting.WithEmptyFilter(),
+				reconcilertesting.WithWebhookForNATS(),
+				reconcilertesting.WithMultipleConditions(),
+				reconcilertesting.WithSinkURLFromSvc(ens.subscriberSvc),
+			},
+			expectBefore: expect{
+				k8sSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.HaveCleanEventTypes(nil),
+					reconcilertesting.HaveCondition(reconcilertesting.DefaultCondition("")),
+					reconcilertesting.HaveSubsConfiguration(configDefault(ens.defaultSubscriptionConfig.MaxInFlightMessages)),
+					reconcilertesting.HaveSubscriptionReady(),
+				},
+			},
+			changeSubscription: func(subscription *eventingv1alpha1.Subscription) {
+				reconcilertesting.AddFilter(reconcilertesting.EventSource,
+					reconcilertesting.OrderCreatedEventTypeNotClean,
+					subscription,
+				)
+			},
+			expectAfter: expect{
+				k8sSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.HaveCondition(reconcilertesting.DefaultCondition("")),
+					reconcilertesting.HaveSubsConfiguration(configDefault(ens.defaultSubscriptionConfig.MaxInFlightMessages)),
+					reconcilertesting.HaveSubscriptionReady(),
+					reconcilertesting.HaveCleanEventTypes([]string{reconcilertesting.OrderCreatedEventType}),
+					gomega.Not(reconcilertesting.HaveCondition(reconcilertesting.MultipleDefaultConditions()[0])),
+					gomega.Not(reconcilertesting.HaveCondition(reconcilertesting.MultipleDefaultConditions()[1])),
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			subscription, subscriptionName := createSubscription(ctx, g, ens, tc.subscriptionOpts...)
+
+			testSubscriptionOnK8s(ctx, g, ens, subscription, subscriptionName, tc.expectBefore.k8sSubscription...)
+			testEventsOnK8s(ctx, g, ens, tc.expectBefore.k8sEvents...)
+			testSubscriptionOnNATS(g, ens, subscriptionName, tc.expectBefore.natsSubscription...)
+
+			tc.changeSubscription(subscription)
+			updateSubscriptionOnK8s(ctx, g, ens, subscription)
+
+			testSubscriptionOnK8s(ctx, g, ens, subscription, subscriptionName, tc.expectAfter.k8sSubscription...)
+			testEventsOnK8s(ctx, g, ens, tc.expectAfter.k8sEvents...)
+			testSubscriptionOnNATS(g, ens, subscriptionName, tc.expectAfter.natsSubscription...)
+			testDeletionOnK8s(ctx, g, ens, subscription, tc.shouldTestDeletion)
```

Comment:
nit: This is a perfect fit for the [GWT pattern](https://github.com/kyma-project/community/blob/main/concepts/eventing-code-guidelines.md#gwt-given-when-then-pattern):

```suggestion
			// given
			subscription, subscriptionName := createSubscription(ctx, g, ens, tc.subscriptionOpts...)

			testSubscriptionOnK8s(ctx, g, ens, subscription, subscriptionName, tc.expectBefore.k8sSubscription...)
			testEventsOnK8s(ctx, g, ens, tc.expectBefore.k8sEvents...)
			testSubscriptionOnNATS(g, ens, subscriptionName, tc.expectBefore.natsSubscription...)
			
			// when
			tc.changeSubscription(subscription)
			updateSubscriptionOnK8s(ctx, g, ens, subscription)

			// then
			testSubscriptionOnK8s(ctx, g, ens, subscription, subscriptionName, tc.expectAfter.k8sSubscription...)
			testEventsOnK8s(ctx, g, ens, tc.expectAfter.k8sEvents...)
			testSubscriptionOnNATS(g, ens, subscriptionName, tc.expectAfter.natsSubscription...)
			testDeletionOnK8s(ctx, g, ens, subscription, tc.shouldTestDeletion)
```

Reply:
added